### PR TITLE
Use Dependencies in Riverpod

### PR DESCRIPTION
**Pull request by bcarterterp**
- Dependencies that were not generated via Riverpod, now are.
- Updated HomeScreen Notifier to be in line with Riverpod 2.0
- Updated HomeScreenStateNotifier tests

### CODE REVIEW

File: lib/presentation/screens/home/notifier/home_screen_state.dart [hunk 1]
```diff
@@ -1,23 +1,27 @@
 import 'package:dio/dio.dart';
+import 'package:equatable/equatable.dart';
 import 'package:flap_app/domain/entity/event.dart';
 import 'package:flap_app/domain/entity/recipe.dart';
 
-class HomePageState {
-  HomePageState({required this.loadRecipesEvent});
+class HomeScreenState extends Equatable {
+  const HomeScreenState({required this.loadRecipesEvent});
```
Comment (cxw320) [inline lib/presentation/screens/home/notifier/home_screen_state.dart:7]: I'm curious - do you have an example to help me understand when we would need to compare 2 instances of the home screen state?

Every time we update values in the state object, we create a new instance - so I'm having trouble coming up with a use case for extending Equatable here.

Comment (cxw320) [reply lib/presentation/screens/home/notifier/home_screen_state.dart:7]: Nice! that makes sense : )))

**Review by cxw320 — APPROVED** [review]
Changes look good to me!

File: lib/presentation/screens/home/view/home_screen.dart [hunk 2]
```diff
@@ -22,34 +23,42 @@ class _HomeScreenState extends ConsumerState<HomeScreen> {
...
-        body: switch (state.loadRecipesEvent) {
-          InitialEvent() => const LoadingScreenWidget(),
-          LoadingEvent() => const LoadingScreenWidget(),
-          SuccessEvent() => RecipeGrid(state: state),
-          EventError() => const ErrorScreenWidget(),
-        });
+      appBar: AppBar(
+        title: const Text('Home Page Screen'),
+      ),
+      body: _getWidget(state.loadRecipesEvent),
```
Comment (amandaroa) [inline lib/presentation/screens/home/view/home_screen.dart:37]: Nice cleanup!

File: test/presentation/screens/home/notifier/home_screen_state_notifier_test.dart [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+      // Call the getRandomRecipes method
+      await container
+          .read(homeScreenStateNotifierProvider.notifier)
+          .getRandomRecipes();
+
+      // Verify that HomePageState has a loadRecipesEvent that is an ErrorEvent
+      final states = stateListener.data;
+      expect(states[0].$2, HomeScreenState.loading());
+      expect(states[1].$2, HomeScreenState.error(errorType));
```
Comment (amandaroa) [inline test/presentation/screens/home/notifier/home_screen_state_notifier_test.dart:101]: What does this $2 mean?

File: test/presentation/screens/home/notifier/home_screen_state_notifier_test.dart [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+        // Call the getRandomRecipes method
+        await container
+            .read(homeScreenStateNotifierProvider.notifier)
+            .getRandomRecipes();
+
+        // Verify that HomePageState is updated with a recipe list
+        final states = stateListener.data;
+        expect(states[0].$2, HomeScreenState.loading());
+        expect(states[1].$2, HomeScreenState.success(recipes));
```
Comment (amandaroa) [inline test/presentation/screens/home/notifier/home_screen_state_notifier_test.dart:62]: This seems to assert a success but the description on line 26 says it is testing the failure state. Am I reading the code wrong?

File: test/presentation/screens/home/notifier/home_screen_state_notifier_test.dart (hunk 3 above)
Comment (amandaroa) [reply test/presentation/screens/home/notifier/home_screen_state_notifier_test.dart:101]: This seems to assert a error but the description on line 67 says it is testing the success state. Am I reading the code wrong?

**Review by amandaroa — APPROVED** [review]
Looks good!

File: test/presentation/screens/home/notifier/home_screen_state_notifier_test.dart (hunk 3 above)
Comment (bcarterterp) [reply test/presentation/screens/home/notifier/home_screen_state_notifier_test.dart]: The title was a goof on my end, and is now fixed. I have made a ListenerData object in place of a tuple which is what caused the $2

File: test/presentation/screens/home/notifier/home_screen_state_notifier_test.dart (hunk 4 above)
Comment (bcarterterp) [reply test/presentation/screens/home/notifier/home_screen_state_notifier_test.dart]: The title was a goof on my end, and is now fixed.

**Pull request merged**